framework/include/utils/MooseUtils.h

```diff
@@ -62,6 +62,11 @@ void escape(std::string & str);
  */
 std::string trim(const std::string & str, const std::string & white_space = " \t\n\v\f\r");
 
+/**
+ * Python like split function for strings.
+ */
+std::vector<std::string> split(const std::string & str, const std::string & delimiter);
```

Comment:
This is a duplication of the `tokenize` function with an independent implementation. I'd prefer to avoid adding this, but if we do, can we at least group it with the others and perhaps implement it in terms of the others?

Reply:
The tokenize doesn't honor empty items, so it would not be possible to pass arguments to certain sub-apps if tokenize is used.

JOIN: "foo;bar;;" -> ["foo", "bar", "", ""]
TOKENIZE: "foo;bar;;" -> ["foo", "bar"]

Reply:
OK - would you mind added that important difference to the doxygen comment?

Reply:
Yes.